src/h5web/visualizations/containers/NxImageContainer.tsx

```diff
@@ -35,9 +39,22 @@ function NxImageContainer(props: VisContainerProps): ReactElement {
     'x',
   ]);
 
-  const value = useDatasetValue(signalDataset.id);
+  const axesLabels = getAxesLabels(entity);
+  const axesIds = axesLabels.reduce<Record<string, HDF5Id>>((acc, axis) => {
+    if (!axis) {
+      return acc;
+    }
 
-  if (!value) {
+    const dataset = getLinkedEntity(entity, metadata, axis);
+    if (dataset && isDataset(dataset)) {
+      acc[axis] = dataset.id;
+    }
+    return acc;
+  }, {});
```

Comment:
Probably worth adding one or more utility functions to reduce duplication between the two containers, since this is pretty much the exact same code.